### PR TITLE
Support For powerMax

### PR DESCRIPTION
**Pull request by rajinir**

### CODE REVIEW

File: src/pilot/deploy-overcloud.py [hunk 1]
```diff
@@ -579,6 +586,14 @@ def main():
         if args.enable_unity_manila:
             env_opts += " -e ~/pilot/templates/unity-manila-config.yaml"
 
+        if args.enable_powermax:
+           if self.settings.powermax_storage_protocol == "iSCSI":
```
Comment (gaelrehault) [inline src/pilot/deploy-overcloud.py]: Deploy-overcloud.py will not resolve self.settings.powermax_storage_protocol, you probably want to pass which protocol to use as another param instead

Comment (rajinir) [reply src/pilot/deploy-overcloud.py]: Fixed it

**Comment by rajinir** [comment]
Tested this on R11 stamp, deployment was successful and sanity test passes

**Comment by gaelrehault** [comment]
Rajini, could you update the sample files Storage Backends comments section to reflect the options ( and the 2 protocoles)

File: src/deploy/osp_deployer/settings/config.py [hunk 2]
```diff
@@ -418,6 +418,28 @@ def __init__(self, settings_file):
...
+        if backend_settings['enable_powermax_backend'].lower() == 'true':
+            self.enable_powermax_backend = True
+            self.powermax_backend_name = backend_settings['powermax_backend_name']
+            self.powermax_san_ip = backend_settings['powermax_san_ip']
+            self.powermax_san_login = backend_settings[
+                'powermax_san_login']
+            self.powermax_san_password = backend_settings[
+                'powermax_san_password']
+            self.powermax_protocol = backend_settings[
+                'powermax_protocol']
```
Comment (gaelrehault) [inline src/deploy/osp_deployer/settings/config.py]: Probably want to check that protocol being set is a valid one. ( especially with mixed case in iSCI)

**Comment by rajinir** [comment]
> Rajini, could you update the sample files Storage Backends comments section to reflect the options ( and the 2 protocoles)

Done.

**Pull request merged**